README.md

```diff
@@ -23,14 +23,149 @@ Or install it yourself as:
 
     $ gem install bitreserve
 
-## Usage
+# Contents
 
-TODO: Write usage instructions here
+* [1. Usage](#1-usage)
+* [2. Authentication](2#-authentication)
+  * [2.1 OAuth2](#21-oauth2)
+  * [2.2 Basic Auth](#21-basic-auth)
+  * [2.4 Personal Access Token](#24-personal-access-token)
+    * [2.4.1 Via argument](#241-via-argument)
+    * [2.4.2 Via environment variable](#242-via-environment-variable)
+* [3. Entities](#3-entities)
+* [4. API calls](#4-endpoints)
+  * [4.1 Currencies](#41-currencies)
+  * [4.2 Tickers](#42-tickers)
+  * [4.4 Entities](#43-entities)
+  * [4.4 Cards](#44-cards)
+  * [4.5 Transactions](#45-transactions)
+  * [4.6 Contacts](#46-contacts)
+  * [4.7 Users](#47-users)
+  * [4.8 Transparency](#48-transparency)
+* [5. Testing](X#-testing)
+* [6. Contributing](X#-contributing)
 
-## Contributing
+# 1 Usage
 
-1. Fork it ( https://github.com/[my-github-username]/bitreserve/fork )
+To start using the gem, you must instantiate the client
+
+```ruby
+client = Bitreserve::Client.new
+```
+
+All subsequent API calls are made from the client instance. For example:
+
+```ruby
+client.me
+# => #<Bitreserve::Entities::User:0x00000002cceff0>
+```
+
+# 2 Authentication
+
+## 2.1 OAuth2
+
+TODO
+
+## 2.2 Basic Auth
+
+TODO
+
+## 2.4 Personal Access Token
+
+If you already have a token, you can use by setting an environment variable, or by passing when instantiating the client
+
+### 2.4.1 Via argument
+
+Pass the token to the constructor:
+
+```ruby
+Bitreserve::Client.new(token: 'your-access-token')
+```
+
+### 2.4.2 Via environment variable
+
+Set the environment variable using [dotenv](https://github.com/bkeepers/dotenv), or by exporting it in your shell:
+
+```bash
+$ export BITRESERVE_AUTH_TOKEN="your-access-token"
+```
+
+Then instantiate the client:
+
+```ruby
+Bitreserve::Client.new
+```
+
+# 3 Entities
+
+TODO
+
+# 4 Endpoints
+
+## 4.1 Currencies
+
+TODO
+
+## 4.2 Tickers
+
+TODO
+
+## 4.3 Entities
+
+TODO
+
+## 4.4 Cards
+
+TODO
+
+## 4.5 Transactions
+
+TODO
+
+## 4.6 Contacts
+
+TODO
+
+## 4.7 Users
+
+
+### Get User
+
+[*Bitreserve documentation on users*](https://developer.bitreserve.org/api/v0/#users)
+
+**Return the details of the current user:**
+
+```ruby
+client.me
+```
+
+**Return value**: `Bitreserve::Entities::User`
```

Comment:
Hum... I would do that as an example. And I wouldn't have an entities section, I'd just defer to the API docs. IMO, we just need to say "hey, this is how to create a client and this is the name of the methods that call the API endpoints".

In situations where it might not be obvious how it works, an example will be nice, other than that will make the README unnecessarily big, I think.